src/librustc_codegen_ssa/mir/debuginfo.rs

```diff
@@ -289,51 +341,51 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 upvars.chain(generator_fields)
             };
 
-            for (variant_idx, field, name, by_ref, ty, var_scope, var_span) in extra_locals {
-                let fields = match variant_idx {
-                    Some(variant_idx) => {
-                        match &closure_layout.variants {
-                            Variants::Multiple { variants, .. } => {
-                                &variants[variant_idx].fields
-                            },
-                            _ => bug!("variant index on univariant layout"),
-                        }
-                    }
-                    None => &closure_layout.fields,
-                };
-
-                // The environment and the capture can each be indirect.
-                let mut direct_offset = Size::ZERO;
-                let indirect_offsets = [
-                    fields.offset(field),
-                    Size::ZERO,
-                ];
-                let mut indirect_offsets = &indirect_offsets[..];
-
-                if !env_ref {
-                    direct_offset = indirect_offsets[0];
-                    indirect_offsets = &indirect_offsets[1..];
+            for (variant_idx, field, name, by_ref, source_info) in extra_locals {
+                let mut proj = env_proj.clone();
```

Comment:
You can use placebuilder here

Reply:
Oh that's a thing now? Last I remembered it was only being discussed.

Reply:
Doesn't seem to be yet.